src/main/scala/com/gu/identity/testing/usernames/TestUsernames.scala

```diff
@@ -1,63 +1,89 @@
 package com.gu.identity.testing.usernames
 
+import com.gu.identity.testing.usernames.TestUsernames.{ConfPayloadByteSize, TestUser, logger, maybeTokenFromEmail, tokenToEmail}
+
 import java.nio.ByteBuffer
 import java.time.Clock.systemUTC
 import java.time.{Clock, Duration, Instant}
 import java.time.Duration.ofMinutes
 import java.time.Instant.ofEpochSecond
-
 import com.typesafe.scalalogging.LazyLogging
 
 
 
-trait TestUsernames {
-  def generate(conf: Array[Byte]): String
+class TestUsernames(usernameEncoder: Encoder, recency: Duration)(implicit clock: Clock) extends LazyLogging {
+  def generate(conf: Array[Byte]): String = {
```

Comment:
I inlined these two functions from the anonymous class and made this trait a class as it didn't seem like the trait was used directly for anything